pkg/util/mount/mount_linux.go

```diff
@@ -727,13 +727,13 @@ func GetSELinux(path string, mountInfoFilename string) (bool, error) {
 
 func (mounter *Mounter) GetMountRefs(pathname string) ([]string, error) {
 	pathExists, pathErr := PathExists(pathname)
-	if !pathExists {
-		return []string{}, nil
-	} else if IsCorruptedMnt(pathErr) {
+	if IsCorruptedMnt(pathErr) {
 		klog.Warningf("GetMountRefs found corrupted mount at %s, treating as unmounted path", pathname)
 		return []string{}, nil
 	} else if pathErr != nil {
 		return nil, fmt.Errorf("error checking path %s: %v", pathname, pathErr)
+	} else if !pathExists {
+		return []string{}, nil
```

Comment:
this code is fine, but I am thinking to simplify it like
if !pathExists {
    //might also log err if err != nil if needed
     return []string{}, err
}
realpath, err := filepath.EvalSymlinks(pathname)
...
return SearchMountPoints(realpath, procMountInfoPath)

Reply:
Current method returns nil if the error is corrupted mount point, and the some places that call this method try to unmount the path when the error is corrupt error(because they don't know about the error). If I make the simplification that you suggest then the caller methods won't try to unmount the path if the error is corrupted.

I think the same relation between `PathExists` and `CleanupMountPoint` is also exist for this method and it's callers.

If I make this simplification then should I change the all the callers of this method to try to unmount even get error from this method?

Reply:
Current method returns nil if the error is corrupted mount point, and the some places that call this method try to unmount the path when the error is corrupt error(because they don't know about the error). If I make the simplification that you suggest then the caller methods won't try to unmount the path if the error is corrupted.

I think the same relation between PathExists and CleanupMountPoint is also exist for this method and it's callers.

If I make this simplification then should I change the all the callers of this method to try to unmount even get error from this method?

Reply:
I think this is different from those teardown functions. This is counting how many valid mount references. If it is a corrupted mount, it will not count as valid mount. There is no unmount function here.

This logic is applicable for the functions that are trying to get mount count, or valid mount etcs. This is not for the function to teardown mounts.

Reply:
Yes you're right but for the unmount part I'm little confused about the usage of this method, for example when I check the usage of this method I find [this place](https://github.com/kubernetes/kubernetes/blob/914f80d671a029c86cc2d5166828c43d4ce6d90d/pkg/volume/util/operationexecutor/operation_generator.go#L933), with the current implementation of the `mount_linux.GetMountRefs` function, If the error is corrupt error, [this line](https://github.com/kubernetes/kubernetes/blob/914f80d671a029c86cc2d5166828c43d4ce6d90d/pkg/volume/util/operationexecutor/operation_generator.go#L933) (`volumeDeviceUmounter.UnmountDevice`) will be executed.

But if I change this method as return error for the corrupt error, then [this line](https://github.com/kubernetes/kubernetes/blob/914f80d671a029c86cc2d5166828c43d4ce6d90d/pkg/volume/util/operationexecutor/operation_generator.go#L939) (`deviceToDetach.GenerateError(`) will be executed.

So the behavior will be different?

Reply:
@jingxu97 Do you have any idea about my concern? Thanks.

Reply:
sorry for the delay. Let me double check this part.

Reply:
@jingxu97 any update on this? Sorry for pinging you again, long live PRs creates extra works :(